code/game/objects/items/RCD.dm

```diff
@@ -729,7 +729,8 @@ RLD
 	if(!range_check(A,user))
 		return
 	if(target_check(A,user))
-		user.Beam(A,icon_state="rped_upgrade",time=30)
+		var/datum/beam/upgradebeam = user.Beam(A,icon_state="rped_upgrade")
+		QDEL_IN(upgradebeam, 3 SECONDS)
```

Comment:
There seems to be a lot of this pattern. Wouldn't it be cleaner to move the QDEL_IN to a part of Beam(), so we don't end up fucking somebody up downline?

Reply:
I'm going to remove all the QDEL_IN and readd the time argument.

The Removal of the Time Argument and its consequences have been a disaster for the pull request.